more block timing fixes

tomjnixon opened the pull request.

tomjnixon commented on ear/fileio/adm/timing_fixes.py:
```diff
@@ -0,0 +1,174 @@
+
+
+def fix_blockFormat_durations(adm):
+    """Modify the diration of audioBlockFormats to ensure that the end of one
```
fixed, thanks

tomjnixon commented on ear/fileio/adm/timing_fixes.py:
```diff
@@ -0,0 +1,174 @@
+        for blockFormat in channelFormat.audioBlockFormats:
+            if (
+                blockFormat.rtime is None
+                or blockFormat.duration is None
```
This function is just for fixing the interpolation length relative to the block extent (i.e. without considering the object extent). This kind of error (where the interpolation length is greater than the object extent and there is no block timing) should probably be handled too, but isn't currently.

Ideally this case wouldn't occur, because interpolation length in a block with no rtime and duration doesn't have any effect (there can be only one such block in a channel, so it is always the first). I'll add a check for this.

benjamin-weiss replied:
Sorry I should have explained that better. What I meant is something like this:

```
<audioBlockFormat audioBlockFormatID="AB_00031001_00000003" rtime="00:00:01.00000">
  <jumpPosition interpolationLength="1.5">1</jumpPosition>
  ...
</audioBlockFormat>
<audioBlockFormat audioBlockFormatID="AB_00031001_00000004" rtime="00:00:02.00000">
  ...
</audioBlockFormat>
```

tomjnixon replied:
ah, ok. That isn't allowed anyway: https://github.com/ebu/ebu_adm_renderer/blob/master/ear/core/renderer_common.py#L246-L255

benjamin-weiss replied:
Ah right, sorry I forgot about that constraint.

tomjnixon commented:
latest version makes sure that the interpolation time is less than the object duration. I decided that the check that rtime and duration are used together should already be done at this point to simplify the logic, so this was moved to the `BlockFormat.verify`. This doesn't add any more requirements on the metadata, just makes a check earlier.

tomjnixon commented:
latest changes:
- rebased onto master
- fixed error where the rtime was interpreted as not being relative to the object start time

to do:
- add CLI for applying to file
- rename underscored classes imported from item selection

tomjnixon commented:
done!

tomjnixon commented:
this generates lots of warnings; we should hide them if there are more than a few so that they don't get lost
